### ISSUE TITLE
minishift  flags --username and --password should not be placed with common flags

### ISSUE THREAD

**Issue by LalatenduMohanty** [minishift/minishift#432]
The general `Flags:` section of minishift usage is for flags which are applicable to all minishift commands. IMO the flags for virtual machine registration is not applicable to all minishift commands.  

```
$ minishift 
Minishift is a command-line tool that provisions and manages single-node OpenShift clusters optimized for development workflows.

Usage:
  minishift [command]

Available Commands:
  config                 Modifies Minishift configuration properties.
  console                Opens or displays the OpenShift Web Console URL.
  delete                 Deletes the Minishift VM.
  docker-env             Sets Docker environment variables.
  get-openshift-versions Gets the list of OpenShift versions available for Minishift.
  ip                     Gets the IP address of the running cluster.
  logs                   Gets the logs of the running OpenShift cluster.
  openshift              Modifies an OpenShift master configuration resource.
  service                Prints the URL for the specified service to the console.
  ssh                    Log in to or run a command on a Minishift VM with SSH.
  start                  Starts a local OpenShift cluster.
  status                 Gets the status of the local OpenShift cluster.
  stop                   Stops the running local OpenShift cluster.
  version                Gets the version of Minishift.

Flags:
      --alsologtostderr                  log to standard error as well as files
      --log_backtrace_at traceLocation   when logging hits line file:N, emit a stack trace (default :0)
      --log_dir string                   If non-empty, write log files in this directory (default "")
      --logtostderr                      log to standard error instead of files
      --password string                  Password for the virtual machine registration.
      --show-libmachine-logs             Show logs from libmachine.
      --stderrthreshold severity         logs at or above this threshold go to stderr (default 2)
      --username string                  User name for the virtual machine registration.
  -v, --v Level                          log level for V logs
      --vmodule moduleSpec               comma-separated list of pattern=N settings for file-filtered logging

Use "minishift [command] --help" for more information about a command.
```

**Comment by hferentschik** [comment]
This is partly true. The idea was that there are several commands which need it, so by putting them globally they can be used by these commands, but yes, it would not apply to all. I am happy to move them to the commands which needs them - _start_, _stop_, _delete_. 

While on it, I think we need to be consistent with the proxy parameters. Either they are global as well, or they should made available for the three commands mentioned above. I can see pro and cons for either approach, global or commands specific options, but either way we need to make things consistent.

**Comment by praveenkumar** [comment]
> While on it, I think we need to be consistent with the proxy parameters. Either they are global as well, or they should made available for the three commands mentioned above. 

Why we need proxy parameters for across all three comands? Since when we do the start it automatic set proxy at VM level which can be used it's lifetime (until it deleted).

**Comment by LalatenduMohanty** [comment]
@praveenkumar Logically if you need proxy parameters for registration then you need them for unregistration too. The current implementation for proxy parameters is not clean as the start command takes  the proxy parameters and somehow we store in the VM but what will happen if we anything other than subscription-manager and it does not store proxy level information in the VM. So we need a consistent implementation.

**Comment by LalatenduMohanty** [comment]
The other issue with registration is that , if you reboot the VM then it will loose the registration information , so we need to store registration information is persistent storage , so that `minishift start` -> rebot -> "yum instal" should work.

**Comment by praveenkumar** [comment]
> Logically if you need proxy parameters for registration then you need them for unregistration too.

Logically we already doing it so when user start the VM using proxy then it set proxy for the VM not only for openshift/docker. you can try to run a vm and then start=>stop=>start it suppose to have proxy already.

> if you reboot the VM then it will loose the registration information , so we need to store registration information is persistent storage , so that minishift start -> rebot -> "yum instal" should work

That is #396, when you say reboot you mean start => stop => start (right) then we already have condition that start parameter always be same one we start the VM so even reboot it shouldn't loose subscriptions.

**Comment by hferentschik** [comment]
> The other issue with registration is that , if you reboot the VM then it will loose the registration as well as proxy information ,

Really? I thought that's what _subscription-manager config _ addresses as well. And anyways, this is assuming you don't pass the parameters again. ATM, we are saying anyways you need to repeat the exact command line on a re-start. Otherwise you are in trouble. For me that falls under #179.

**Comment by LalatenduMohanty** [comment]
> That is #396, when you say reboot you mean start => stop => start (right) then we already have condition that start parameter always be same one we start the VM so even reboot it shouldn't loose subscriptions.

Thats not user friendly at all.

**Comment by praveenkumar** [comment]
> Thats not user friendly at all.

Yes but we don't have any option right now until we break start command (because it manage 2 different set of stuff) with create and then start which we discussed many time and part of #179

**Comment by LalatenduMohanty** [comment]
> That is #396, when you say reboot you mean start => stop => start (right) then we already have condition that start parameter always be same one we start the VM so even reboot it shouldn't loose subscriptions.

@praveenkumar @hferentschik The issue is little different than #179 here. The host reboot causes the rootfs to go back to original state, so I am not sure if unregister actually removes the entitlement clearly.

**Comment by praveenkumar** [comment]
> I am not sure if unregister actually removes the entitlement clearly.

If a user try do start => stop => start then in the stop time it is unregister which is clear (please try it once). we have unregister hooked with stop and delete command.

**Comment by hferentschik** [comment]
> The issue is little different than #179 here. 

Yes and no. I expect that we end up with something where we specify parameters in start/create, but then in successive restart we don't have to specify anything. In this case, also the registration parameters would be re-applied. 

> The host reboot causes the rootfs to go back to original state, so I am not sure if unregister actually removes the entitlement clearly.

Seems to me that either we make sure that the registration state persists or we don't use _subscription-manager config _ and go back to what we had.

**Comment by LalatenduMohanty** [comment]
@praveenkumar Nope, thats not the point .We have to cross check that when we register the host, an entitlement given to an user ( I can see them  by logging to access.redhat.com ) but when we unregister if the entitlement is removed. At this moment I have so much entitlements that I cannot verify if the entitlement is removed.

**Comment by praveenkumar** [comment]
> We have to cross check that when we register the host, an entitlement given to an user ( I can see them by logging to access.redhat.com ) but when we unregister if the entitlement is removed. 

This is already what I tested when implemented register/unregister, I am not sure why so much confusion around it.

**Comment by LalatenduMohanty** [comment]
> This is already what I tested when implemented register/unregister, I am not sure why so much confusion around it.

@praveenkumar May be I am the only person confused here. So I am assuming that you have tested that entitlement is removed properly if I unregister after reboot , right?

**Comment by praveenkumar** [comment]
> I expect that we end up with something where we specify parameters in start/create, but then in successive restart we don't have to specify anything. In this case, also the registration parameters would be re-applied.

Agree

> Seems to me that either we make sure that the registration state persists or we don't use _subscription-manager config _ and go back to what we had.

Right now we are not using persistence subscription-manager config_ (or we do?) I think that is part of #396

**Comment by praveenkumar** [comment]
> So I am assuming that you have tested that entitlement is removed properly if I unregister after reboot , right?

if here reboot you means start => stop => start then entitlement is removed properly (during stop) and then again attached properly (during start).

**Comment by LalatenduMohanty** [comment]
> Seems to me that either we make sure that the registration state persists or we don't use _subscription-manager config _ and go back to what we had.

@hferentschik These are two different things IMO. We do not have persistence config because we use live-rw filesystem which does not persists reboots.

**Comment by LalatenduMohanty** [comment]
> if here reboot you means start => stop => start then entitlement is removed properly (during stop) and then again attached properly (during start).

I mean start -> `minishift ssh` -> reboot -> unregister

**Comment by praveenkumar** [comment]
> I mean start -> minishift ssh -> reboot -> unregister

Why would you reboot after ssh to minishift? In that case you not even get openshift running after reboot. I don't think that is a valid case @hferentschik ?

**Comment by LalatenduMohanty** [comment]
@hferentschik @praveenkumar it is a valid user case, may be not a high priority. But we have all the commands for the users to reboot if they are are experimenting with the VM. Also some user tend to shutdown VMs to save resources.

**Comment by LalatenduMohanty** [comment]
The important point is entitlements are expensive and so it should not be wasted.

**Comment by praveenkumar** [comment]
@LalatenduMohanty  for me valid use case for a user if want to save state/resource then `minishift stop` not do ssh and reboot from there (that doesn't save the state/resource). I agree that entitlement is expensive and we should use it carefully.

**Comment by hferentschik** [comment]
Merged via pull request #483.
